Simplify LICENSE file, add LICENSE.header and add AUTHORS.

mkrautz opened the pull request:
This simplifies the LICENSE file refer to the copyright
holders of Mumble as "The Mumble Developers". The client
already does this in the About dialog.

The entity "The Mumble Developers" is the name we use for
the copyright holders of Mumble. These are listed in the
AUTHORS file.

The AUTHORS file is generated via scripts/generate-AUTHORS.py.
The script looks at the Git history and removes duplicates and
other mistakes made through the years.

All new files in the repo should use the license header found
in LICENSE.header.

The AUTHORS and LICENSE files are permalinked to

    https://www.mumble.info/LICENSE
    https://www.mumble.info/AUTHORS

These locations are used in the files themselves,
as well as the license header.

mkrautz commented:
See also

https://github.com/mumble-voip/mumble/issues/1159
https://github.com/mumble-voip/mumble/issues/1513

mkrautz commented:
/cc @hacst @Kissaki 

Also curious to hear what @thorvald has to say.

mkrautz commented:
Should we add some sentences to the AUTHORS file describing that authorship history for copyright purposes is available in the project's Git history -- just to spell it out.

mkrautz commented:
It would be nice if we could change the license header of our existing files, but we would need an OK from the authors of those files to change the license header.

If we can get an OK from @thorvald to do that, most files should be covered, I think!

mkrautz commented:
One thing that would be nice to be able to do, is to remove the copyright year range from
LICENSE.header, and just say:

    // Copyright The Mumble Developers. All rights reserved.
    // [...]

But opinions on whether that's a good idea differ. My idea for now is to leave it in, and bump the end of the range when a change is made to the file.

thorvald commented:
I love this initiative. Open Source is the donation of our most precious commodity; time, and one of the ways we can give back is proper attribution and recognition. This is something I deeply believe in, and this change prominently recognizes all contributors.

Now, removing aliases makes sense, but right now you remove all my aliases, so I'm not listed in the resulting AUTHORS file at all. Ahem :)

Since we no longer have emails of individuals in the license, it might be a good idea to list a contact method for licensing questions -- a developer-only mailing list for example, or a private issue here on github?

Also, make sure the AUTHORS file is distributed along with every binary that we produce, and encourage repackagers to do the same -- the binaries might outlive domain names.

mkrautz commented:
@thorvald 

Oops! I'll double check the blacklist in the script to ensure everyone is included.

I'll make sure the list of authors is included in the binaries. For Mumble, it'll be in the about dialog somewhere.

For Murmur, I think we should add a "murmurd -license" option to view the license information. And probably add something like:

    Murmur 1.3.0
  
    Copyright 2005-2016 The Mumble Developers. All rights reserved.
    Use of this program is governed by a BSD-style license.
    See <https://www.mumble.info/LICENSE> or pass the -license
    parameter to Murmur for more information.

    $REST_OF_USAGE

for Murmur's "usage" text.

As for the license contact, we could do a "licensing@mumble.info", but that could be taken the wrong way. It could make people think we offer special commercial licensing options. But I think it does make sense to have a contact.

What do you think about adding the following line at the very bottom of the LICENSE file of this PR?

    For questions and inquiries about Mumble's license, please contact <license@mumble.info>.

mkrautz commented:
@thorvald 

Also, what do you think about changing the existing license headers of files? We would have to contact contributors to do it (to obey the BSD license), but most of the files contain *your* copyright.
So having your blessing (along with the other regular contributors) would mean we can change a lot of the files already.

mkrautz commented on AUTHORS:
```diff
@@ -0,0 +1,176 @@
+//
+// The use of Mumble source code is governed by a BSD-style
+// license that can be found in the LICENSE file at the root
+// of the Mumble source tree or at <http://www.mumble.info/LICENSE>.
```
https instead of http.

mkrautz commented on AUTHORS:
```diff
@@ -0,0 +1,176 @@
+stevenh <steven.hartland@multiplay.co.uk>
+SuperNascher <kevin@nascher.org>
+Sven-Hendrik Haase <sh@lutzhaase.com>
+Svenne33 <svenne33@users.sourceforge.net>
```
Missing slicer.

mkrautz commented on LICENSE.header:
```diff
@@ -0,0 +1,4 @@
+// Copyright 2016 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <http://www.mumble.info/LICENSE>.
```
https instead of http

mkrautz commented on scripts/generate-AUTHORS.py:
```diff
@@ -0,0 +1,199 @@
+# Copyright 2016 The Mumble Developers. All rights reserved.
+# Use of this source code is governed by a BSD-style license
+# that can be found in the LICENSE file at the root of the
+# Mumble source tree or at <http://www.mumble.info/LICENSE>.
```
https instead of http.

thorvald commented:
"For questions and inquiries" LGTM

Per-file contributors... Yes, you have my blessing.

That said, it's a difficult topic; the per-file header is a way to recognize the amount of contribution done, but it never really did an accurate job of doing so, and it is a PITA to maintain. The "core" of developers will forever be associated with this project, but for contributed patches and pull requests, we do have some contributions of a few lines, and some of a few hundred lines, and they get equal recognition right now.

Perhaps we could list number of commits or number of lines changed in the authors file? That still wouldn't be very accurate (how many lines of code is a SVG icon?), but at least there would be something.

I don't know what a good way to do that is; I feel we owe it to the people who contribute to not only recognize that they did contribute, but also recognize how much... But I have no idea how exactly to do that.

Feel free to disentangle those issues though; update the headers first, then if at some point in the future we find a reasonable and fair way to augment the authors file, we can do so :)

mkrautz commented:
Quoting myself:

>Should we add some sentences to the AUTHORS file describing that authorship history for copyright  purposes is available in the project's Git history -- just to spell it out? I think so. 

I think we should add this above the people listing in AUTHORS:

    Mumble's code is developed in a Git repository. A log of every contribution ever made to Mumble is available in the Git repository. The Git repository This makes the Git repository the canonical source of contributions to Mumble from an attribution and copyright perspective.

mkrautz commented:
Quoting myself:

>Should we add some sentences to the AUTHORS file describing that authorship history for copyright  purposes is available in the project's Git history -- just to spell it out? I think so. 

I think we should add this above the people listing in AUTHORS:

    Mumble's code is developed in a Git repository. A log of every contribution
    ever made to Mumble is available in the Git repository. The Git repository
    can be queried to get detailed authorship information for copyright and attribution
    purposes for each file that makes up the software.
    A detailed analysis of contributions made to Mumble is available via GitHub's
    contribution statistics: https://github.com/mumble-voip/mumble/graphs/contributors

mkrautz commented:
@thorvald Would that satisfy the need have a measure of each contributor's contribution to the project?

thorvald commented:
I believe it will.

Kissaki commented:
LGTM
suggestions also fine with me

mkrautz commented on AUTHORS:
```diff
@@ -0,0 +1,176 @@
+Justin McGrath <justin.m.mcgrath@gmail.com>
+Karl Dietz <dekarl@users.sourceforge.net>
+karokatona <karokatona@users.sourceforge.net>
+Kevin Rohland <kevin@nascher.org>
```
Alias of Nachser?

mkrautz commented on AUTHORS:
```diff
@@ -0,0 +1,176 @@
+Stefan Hacker <dd0t@users.sourceforge.net>
+Steve Hill <github@cheesy.sackheads.org>
+stevenh <steven.hartland@multiplay.co.uk>
+SuperNascher <kevin@nascher.org>
```
Alias of Nachser?

actown commented on AUTHORS:
```diff
@@ -0,0 +1,176 @@
+d-rez <dark.skeleton@gmail.com>
+dc6jgk <github.filter@gkware.com>
+dennisschagt <dennisschagt@gmail.com>
+Derrick Dymock <derrick@puppetlabs.com>
```
actown@gmail.com please

hacst commented on scripts/generate-AUTHORS.py:
```diff
@@ -0,0 +1,199 @@
+
+	f = codecs.open("AUTHORS", "w", "utf-8")
+	f.write('''// This is the official list of people who have contributed
+// to, and hold the copyright to Mumble.
```
"and/or hold the copyright" maybe? IANAL but afaik not every contribution gives you copyright and not all who hold some copyright do so because they contributed code (e.g. the bonjour code we built on).

Or maybe just strike the copyright part completely. Dunno...

hacst commented:
Yay for simplifying that stuff (though I remember the last time I tried to extract attribution from commit history to fix the woefully incomplete and outdated per file headers @thorvald was appalled I would even consider such a thing ;) ).

One thing to be aware of is that some contributors might be only in the per-file licensing headers or only mentioned by name in early commit messages in SVN before the switch to git. We might want to emphasize that the list might be incomplete and the repository history (including comments and historic file contents) is to be considered the authorative source of per-file authorship information. Then there's transifex which we don't pull contributor information from at this point either...

Apart from "volume" another possible way to emphasize continued contributions is to list the years and year-ranges of contributions and sort by year of last contribution and then duration. It's pretty close to how you usually handle it in the per file licensing headers and quickly gives and overview of the current and long-term contributors to the project. No idea how well it works with so many people. Also substantial one time contributions obviously aren't visible.

As to the notion of "core" developers. It is useful for "representing" the project I guess. Not sure if we need to make it visible in the AUTHORS file. Definitely a special mention of @thorvald is in order for starting the project though ;)

xPaw commented on scripts/generate-AUTHORS.py:
```diff
@@ -0,0 +1,199 @@
+import codecs
+import locale
+
+blacklist = (
```
Why not use `.mailmap` for this? https://git-scm.com/docs/git-shortlog#_mapping_authors

mkrautz replied:
Thanks for the tip! I think this makes sense.

However, I think we should land with the current script as-is. Then we can port it to use .mailmap in a follow-up.

mkrautz commented:
v2 @ https://github.com/mkrautz/mumble/commit/90dcb89c659fca5d6aad69252cb7081f0fd3396b

mkrautz commented:
- Fixed http -> https.
- Added "and/or hold the copyright" to AUTHORS
- Added GitHub statistics link and text about "Mumble's code is developed in a Git repository [...]"
- Changed "git log --all" to be "git log master 1.2.x"; the old method took all my branches into account...

mkrautz commented:
- Added special thanks to @thorvald  :-)  I added it at the bottom of the file, because that was what I first thought of. Maybe it is more appropriate in the header...

mkrautz commented:
I think this is pretty much ready to land, but I will let this sit until tomorrow evening CET.

One thing I will add tomorrow is credit for patches applied back in the SVN days.

If no one else has any issues, I will land tomorrow.

thorvald commented:
Ah. Is this the right branch? Your personal repo has a license-tweaks-v2, but this pull is the v1 one.

mkrautz commented:
The one from my v2 branch is correct (I linked it earlier).

I typically link to the new version in the PR to have multiple revisions available for viewing, in order to keep comments, and so on, intact.

Sorry for the confusion.

mkrautz commented:
No changes today, I will try again tomorrow.

mkrautz commented:
v3 @ https://github.com/mkrautz/mumble/commit/9bae8799956433465ccfee22de708513178fe11f

mkrautz commented:
v4 @ 1d732fedd80c7c3596213a0b7c0651c9fe6f2f73

mkrautz commented:
I think this is ready to land. Will land tomorrow unless there are any comments!

mkrautz commented:
Landed as 3b2a169f463bb3e83d8787951dd0639e12f12306.
